sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/DefaultScheduler.java

```diff
@@ -587,12 +600,22 @@ protected DefaultScheduler(
         this.recoveryPlanOverriderFactory = recoveryPlanOverriderFactory;
         this.failurePolicyOptional = serviceSpec.getReplacementFailurePolicy();
         this.updateResult = updateResult;
+        processOffers();
     }
 
     @VisibleForTesting
-    void awaitTermination() throws InterruptedException {
-        executor.shutdown();
-        executor.awaitTermination(AWAIT_TERMINATION_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+    void awaitOffersProcessed() throws InterruptedException {
+        while (!offerQueue.isEmpty()) {
+            LOGGER.warn("Queue is NOT empty");
+            Thread.sleep(1000);
+        }
+
+        while (!offersInProgress.isEmpty()) {
```

Comment:
There is a window/race between grabbing the offers from offerQueue (L782) and adding them to offersInProgress (L788) in which both lists would be empty.